osu.Game/Screens/Play/Player.cs

```diff
@@ -962,7 +962,7 @@ public override bool OnExiting(IScreen next)
             screenSuspension?.Expire();
 
             // if arriving here and the results screen preparation task hasn't run, it's safe to say the user has not completed the beatmap.
-            if (prepareScoreForDisplayTask == null)
+            if (Score != null && prepareScoreForDisplayTask == null)
```

Comment:
A little unfortunate that this was added, I presume due to changes in `MultiplayerPlayer`? If that implementation is potentially tolerant of null score, then maybe `OnExiting()` should be overridden and the null checked there locally? Or is there some other particular reason for null-checking in the base implementation? (aside from being a catch-all for any future implementations, I suppose).

Reply:
It's part of a commit to stop `MultiplayerPlayer` crashing when a beatmap can't be loaded (see https://github.com/ppy/osu/pull/14179/commits/fcec714b4f545ccc5fbddd4d21a61508f5b7bc62). I'll loop back on why this particular change is required.. maybe it can be replaced with another `LoadedBeatmapSuccessfully` call if that sits better?

Reply:
If such a check would be local to the multiplayer player, then yeah. My primary concern is that the base `Player` in general doesn't really ever null-check `Score` other than this one special case, so a few months down the line this check which is understandable in the context of this pull could become a headscratcher (or even start triggering inspections).

Reply:
Hmm, I think the base `Player` actually needs this check though?

https://github.com/ppy/osu/blob/fcec714b4f545ccc5fbddd4d21a61508f5b7bc62/osu.Game/Screens/Play/Player.cs#L159-L162

Reply:
I suppose the lack of such a check must have been a latent bug rather than an issue caused by the multiplayer implementation specifically, then? As in I'd expect it to affect _any_ player implementation.

If the above understanding is correct, then I have no issues, although that might also warrant further examination of all of the other usages of `Score` in player...?

Reply:
Hmm... if i remember correctly i ran into this with some very weird testing setup. I'd argue that it can't happen in normal flows due to the screen not being `Entered` when the load is not successful.

I'll remove this for now as I don't think it's required in this PR>